_ingest-pipelines/processors/dot-expander.md

```diff
@@ -0,0 +1,303 @@
+
+```json
+PUT /_ingest/pipeline/dot-expander
+{
+  "description": "Dot expander processor",
+  "processors": [
+    {
+      "dot_expander": {
+        "field": "user.address.city"
+      }
+    },
+    {
+        "dot_expander":{
+         "field": "user.address.state"
+        }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 2 (Optional): Test the pipeline
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/dot-expander/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "field": "city, state"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "field": "city, state"
+        },
+        "_ingest": {
+          "timestamp": "2023-11-17T23:49:27.597933805Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+### Step 3: Ingest a document
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=dot-expander
+{
+  "field": "Denver, CO"
+}
+```
+{% include copy-curl.html %}
+
+### Step 4 (Optional): Retrieve the document
+
+To retrieve the document, run the following query:
+
+```json
+GET testindex1/_doc/1
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following response confirms the document was indexed:
+
+```json
+{
+  "_index": "testindex1",
+  "_id": "1",
+  "_version": 58,
+  "_seq_no": 57,
+  "_primary_term": 30,
+  "found": true,
+  "_source": {
+    "field": "Denver, CO"
+  }
+}
+```
+
+## Nested fields
+
+The `dot_expander` processor consolidates the `user.address.city` and `user.address.state` fields by merging with an existing `address`, `city`, and `state` field nested under `user`. If the field is a scalar value, then it will turn that field into an array. Take for example the following document:
+
+```json
+{
+    "dot_expander": {
+        "field": "user.address.city",
+        "override": true
+    }
+}
+```
+
+With `override:false` (or unset), the dot-pathed top-level field gets appended to the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": ["Denver", "Boulder"],
+            "state": "CO"
+        }
+    }
+}
+```
+
+With `override:true`, the dot-pathed top-level field value overwrites the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": "Boulder",
+            "state": "CO"
+        }
+    }
+}
+```
+
+If the field value is set to a wildcard `*`, the processor expands all top-level dotted field names, for example:  
```

Comment:
Please remove. OpenSearch does not support wildcards in field names.

Reply:
Removed